src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala

```diff
@@ -15,6 +15,8 @@ trait Browser {
   def parseFile(path: String, charset: String): Document = parseFile(new File(path), charset)
   def parseFile(path: String): Document = parseFile(new File(path), "UTF-8")
 
+  def parseResource(name: String, charset: String = "UTF-8"): Document = parseFile(new File(getClass.getResource(name).toURI), charset)
```

Comment:
It'd probably be something like `parseString` - building a `WebResponse` object and loading it into a window. `WebResponse` constructors receive a `WebResponseData` object containing the page contents.

To build a `WebResponseData` you can either read all the stream into a `byte[]` using something like `IOUtils.readFully` and pass it to the constructor or create your own instance of `DownloadedContent` using directly the input stream. I'd be ok with either approach.